### DISCUSSION TITLE
environment variable does not exist

### DISCUSSION THREAD

**Discussion by henry-ht** [imdhemy/laravel-in-app-purchases#139] [Q&A]
just declaring the environment variable GOOGLE_APPLICATION_CREDENTIALS in the .env file is not enough for the Google Auth Library for PHP to use it because it is null, reading the documentation of the previous plugin the only way to declare that environment variable is with the following code:

`putenv('GOOGLE_APPLICATION_CREDENTIALS=/path/to/my/credentials.json');`

[https://github.com/googleapis/google-auth-library-php](url)

After adding this line of code in a middleware, the plugin was able to give an affirmative answer.

It is necessary to document this or create a function that adds this variable so that it is accessible by the Google Auth Library for PHP.

**Comment by imdhemy** [comment]
@henry-ht 

I created issue #140  from this discussion. Checking the [Laravel documentation](https://laravel.com/docs/configuration#environment-configuration), it appears to me that declaring the `.env` variable is enough. 

I also tried with tinker as follows:

```shell
Psy Shell v0.11.8 (PHP 8.1.10 — cli) by Justin Hileman

>>>  getenv("GOOGLE_APPLICATION_CREDENTIALS")
=> "/path/to/my/credentials.json"
```

What do you think?

**Reply by imdhemy** [reply]
@henry-ht 👋
